### DISCUSSION TITLE
How can I remove an item or a property that is not usefull anymore?

### DISCUSSION THREAD

**Discussion by versant2612** [LeMyst/WikibaseIntegrator#355] [Q&A] [answered]
I have a local wikibase instance and I am building a bulk load program. 

How can I remove an item or a property that is not usefull anymore?

**Comment by LeMyst** [comment, marked as answer]
Hello @versant2612 
If you use the pre-release version (0.12), you can do something like this:
```
login_instance = <login instance creation>
wbi = WikibaseIntegrator(login=login_instance)
item = wbi.item.get('Q1234')
item.delete()

prop = wbi.property.get('P1234')
prop.delete()
```

**Reply by versant2612** [reply]
I got an error when deleting

Traceback (most recent call last):
  File "/home/veronicasantos/loadNDTLD.py", line 213, in <module>
    createEntity(root)
  File "/home/veronicasantos/loadNDTLD.py", line 188, in createEntity
    del_item.delete()
  File "/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/entities/baseentity.py", line 277, in delete
    raise ValueError("The entity must have a pageid attribute correctly set")
**ValueError: The entity must have a pageid attribute correctly set**

**Reply by LeMyst** [reply]
Hello @versant2612 ,
It's fixed in c5bb9f65f48aa474d4bfa5fa2cb8c7c123d8f606
You can update your library from the git.

You can use this command:
`python -m pip install git+https://github.com/LeMyst/WikibaseIntegrator.git@rewrite-wbi#egg=wikibaseintegrator`
or add this to your requirements.txt:
`git+https://github.com/LeMyst/WikibaseIntegrator.git@rewrite-wbi#egg=wikibaseintegrator`

Thank you for your report.

**Reply by versant2612** [reply]
It is working. Thank you for the fix.

**Comment by versant2612** [comment]
Thank you @LeMyst .

Since I don't have the QID, I will need to search by entity name or property name in order to get it, right?

**Reply by LeMyst** [reply]
You can use the `search_entities` function from the `wbi_helpers` module.
Doc: https://wikibaseintegrator.readthedocs.io/en/rewrite-wbi/wikibaseintegrator.wbi_helpers.html#wikibaseintegrator.wbi_helpers.search_entities

**Comment by versant2612** [comment]
Is there a way to delete a group of items based on the type and the user who created it, like this operation we do through the UI?

![image](https://user-images.githubusercontent.com/46487949/174664304-c1736b8d-0d51-4c14-85b9-c15385343162.png)

**Reply by LeMyst** [reply]
Maybe it's possible to do it from the API, and WikibaseIntegrator allow to easily access the API with `wbi_helpers.mediawiki_api_call_helper()`.
(But I don't think it's possible from the API)

But there will be no special function inside WBI to do it, because it's not a Wikibase related need, only a MediaWiki ones.

**Reply by dpriskorn** [reply]
The way I do it is getting the list of items via sparql and deleting them one at a time. It's slow but works.

**Reply by versant2612** [reply]
@dpriskorn I'm already doing it one at a time, but it's too slow.
Using the UI I can select items or properties that were inserted by a bot user and delete.
As I am loading a large number of elements and I often need to reprocess the dataset, a batch delete function would be useful.

**Reply by dpriskorn** [reply]
I completely understand. I have the same issue. Would you be willing to open a new ticket [here](https://phabricator.wikimedia.org/project/view/17/)?

**Reply by LeMyst** [reply]
I think one answer would be to bypass the Mediawiki API and directly call the Mass delete page with a correctly formatted form (with requests python package).

**Reply by versant2612** [reply]
> I completely understand. I have the same issue. Would you be willing to open a new ticket [here](https://phabricator.wikimedia.org/project/view/17/)?

@dpriskorn I found this [old ticket](https://phabricator.wikimedia.org/T35470) and added a comment.
